rust-runtime/aws-smithy-async/src/rt/sleep.rs

```diff
@@ -39,15 +39,44 @@ where
     }
 }
 
+/// Wrapper type for sharable `AsyncSleep`
+#[derive(Clone, Debug)]
+pub struct SharedAsyncSleep(Arc<dyn AsyncSleep>);
+
+impl SharedAsyncSleep {
+    /// Create a new `SharedAsyncSleep` from `AsyncSleep`
+    pub fn new(sleep: impl AsyncSleep + 'static) -> Self {
+        Self(Arc::new(sleep))
+    }
+}
+
+impl AsRef<dyn AsyncSleep> for SharedAsyncSleep {
+    fn as_ref(&self) -> &(dyn AsyncSleep + 'static) {
```

Comment:
What is the `+ 'static` intended for here?

Reply:
The short answer is it's required by the compiler. My interpretation is that `dyn AsyncSleep` part is derived from the underlying `Arc` so needs to be annotated as being able to live as long as `'static`.